File: 0905-Sort_Array_By_Parity/Sort_Array_By_Parity.go
package sortarraybyparity

func sortArrayByParity(nums []int) []int {
	newArr := make([]int, len(nums))

	left := 0
	right := len(nums) - 1

	for i := range nums {
		if nums[i]%2 == 1 {
			newArr[right] = nums[i]
			right--
		} else {
			newArr[left] = nums[i]
			left++
		}
	}

	return newArr
}

File: 0905-SortArrayByParity/SortArrayByParity.java

class SortArrayByParity {
    class Solution {
        public int[] sortArrayByParity(int[] nums) {
            int[] newArr = new int[nums.length];
            
            int left = 0;
            int right = nums.length - 1;
            for(int i = 0; i < nums.length; i++){
                if(nums[i] % 2 == 1){
                    newArr[right--] = nums[i];  
                } else{
                    newArr[left++] = nums[i];
                }
            }
            
            return newArr;
        }
    }
}
